pkg/plugins/policies/meshtrace/plugin/v1alpha1/plugin.go

```diff
@@ -137,6 +145,52 @@ func applyToOutbounds(
 	return nil
 }
 
+func applyToRealResources(
+	ctx xds_context.Context,
+	rules core_rules.SingleItemRules,
+	rs *xds.ResourceSet,
+	proxy *xds.Proxy,
+) error {
+	for uri, resType := range rs.IndexByOrigin(xds.NonMeshExternalService) {
+		destination, err := getDestinationName(uri, ctx)
+		if err != nil {
+			return err
+		}
+		for typ, resources := range resType {
+			switch typ {
+			case envoy_resource.ListenerType:
+				for _, listener := range resources {
+					if err := configureListener(rules, proxy.Dataplane, listener.Resource.(*envoy_listener.Listener), destination); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getDestinationName(uri model.TypedResourceIdentifier, ctx xds_context.Context) (string, error) {
```

Comment:
is it possible to use https://github.com/kumahq/kuma/blob/882a878cdc0495fccef5300bc107a035bb22cf60/pkg/plugins/policies/core/xds/meshroute/listeners.go#L189 instead?